messaging/kafka/handler.go

```diff
@@ -67,14 +67,12 @@ func Handle(c <-chan kafka.Message) (<-chan []kafka.Message, <-chan bool) {
 			return
 		}
 
-		if len(messages) > 0 {
-			out <- buildKafkaMessages(messages)
-		}
+		buildKafkaMessages(messages, out)
 	}
 
 	go func() {
 		for message := range c {
-			go handler(message)
```

Comment:
remove async message processing. streaming brokers are FIFO and changing that would break the user's expectations.